src/libstore/realisation.cc

```diff
@@ -140,6 +140,12 @@ StorePath RealisedPath::path() const {
     return std::visit([](auto && arg) { return arg.getPath(); }, raw);
 }
 
+bool Realisation::isCompatibleWith(const Realisation & other) const
+{
+    assert (id == other.id);
+    return outPath == other.outPath;
```

Comment:
```suggestion
    return outPath == other.outPath;
    assert(dependentRealisations == other.dependentRealisations);
```

(just to make sure that we keep the invariant)

Reply:
29fddfdf4